File: frontend/ase-delivery-portal/src/repository/types/delivery.ts
export enum DeliveryStatus {
  CREATED = "CREATED",
  EN_ROUTE = "EN_ROUTE",
  DELIVERED = "DELIVERED",
  COMPLETED = "COMPLETED",
}

export interface Delivery {
  id: string;
  status: DeliveryStatus;
  courierEmail: string;
  customerEmail: string;
  boxId: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface DeliveryWithTrackingUrl {
  delivery: Delivery;
  trackUrl: string;
}

export interface CreateDeliveryResponse {
  delivery: Delivery;
  trackUrl: string;
}

export interface DeliveryUpdate {
  id: string;
  courierEmail: string;
  customerEmail: string;
  boxId: string;
}

export interface DeliveryInput {
  courierEmail?: string;
  customerEmail?: string;
  boxId?: string;
}

File: backend/delivery-service/src/main/java/com/asedelivery/deliveryservice/model/Delivery.java
package com.asedelivery.deliveryservice.model;

import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Getter
@Setter
@Builder
@Document(collection = "deliveries")
public class Delivery {

    @Id
    private String id;

    @NonNull
    private DeliveryStatus status;

    @NonNull
    private String courierEmail;

    @NonNull
    private String customerEmail;

    @NonNull
    private String boxId;

    @CreatedDate
    private Date createdAt;

    @LastModifiedDate
    private Date updatedAt;

}
